### PR TITLE
Added data for RegExp.prototype.hasIndices

### PR DESCRIPTION
**Pull request by SebastianZ**
This patch adds compatibility data for the [recently introduced `RegExp.prototype.hasIndices`](https://github.com/tc39/proposal-regexp-match-indices) property.

It got implemented in Firefox 88 as of [bug 1519483](https://bugzilla.mozilla.org/show_bug.cgi?id=1519483) and in Chrome as of [issue 9548](https://bugs.chromium.org/p/v8/issues/detail?id=9548), which should be 91. Though the [platform status](https://www.chromestatus.com/features/6558676666023936) says "In developer trial (Behind a flag)", which confuses me a little bit because it's not needed to turn on any flag.

I've tested this in Firefox Nightly 88.0a1 (2021-03-17), in Chrome 91.0.4450.0. I've also tried Firefox DevEdition 87.0b9 (64-bit) and Chrome 89.0.4389.90 where it is _not_ implemented.

Note that I _didn't_ test the feature in Chrome 88. It might be that it's already added in that version. I also did _not_ test it in other browsers but I assume it's not implemented there yet as the feature is pretty new. The feature just landed in V8 last month, so Chromium based browsers probably didn't pick it up yet. Therefore I've set everything to `false`. I could also set the ones where support is unclear to `null`, instead.

Sebastian

### CODE REVIEW

**Review by Elchi3 — CHANGES_REQUESTED** [review]
Thanks for your PR!
Please run `npm run fix` to sort in the new data correctly per the CI failure: 


```
Problems in 1 file:
2346
✖ javascript/builtins/RegExp.json
2347
  Style – 1 error:
2348
  → Feature sorting error on line #429
2349
    Actual:           "ignoreCase": {
2350
    Expected:         "hasIndices": {
2351
    Tip: Run npm run fix to fix sorting automatically
```

**Comment by SebastianZ** [comment]
Thank you @Elchi3. Ran `npm run fix` now and pushed the changes.

Sebastian

**Comment by Elchi3** [comment]
Looks good to me now structurally. 
I think @jpmedley should comment on the Chrome data here. Looks like it's a bit early to tell. I can't confirm 91.

File: javascript/builtins/RegExp.json [hunk 1]
```diff
@@ -426,6 +426,57 @@
             }
           }
         },
+        "hasIndices": {
+          "__compat": {
+            "mdn_url": "https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/RegExp/hasIndices",
+            "support": {
+              "chrome": {
+                "version_added": "91"
```
Comment (ddbeck) [inline javascript/builtins/RegExp.json:434]: I was wondering how this was coming along and I couldn't figure it out from the bug links. I tested it though and found that it's available in 90 (beta now, but stable in a few days), not 91.

```suggestion
                "version_added": "90"
```

File: javascript/builtins/RegExp.json [hunk 2]
```diff
@@ -426,6 +426,57 @@
...
+              },
+              "chrome_android": {
+                "version_added": "91"
+              },
+              "edge": {
+                "version_added": false
```
Comment (ddbeck) [inline javascript/builtins/RegExp.json:440]: It also works in Edge 90

```suggestion
                "version_added": "90"
```

File: javascript/builtins/RegExp.json [hunk 3]
```diff
@@ -426,6 +426,57 @@
...
+              },
+              "safari_ios": {
+                "version_added": false
+              },
+              "samsunginternet_android": {
+                "version_added": false
+              },
+              "webview_android": {
+                "version_added": false
```
Comment (ddbeck) [inline javascript/builtins/RegExp.json:470]: No reason to expect WebView to omit this:

```suggestion
                "version_added": "90"
```

File: javascript/builtins/RegExp.json [hunk 4]
```diff
@@ -426,6 +426,57 @@
...
         },
+        "hasIndices": {
+          "__compat": {
+            "mdn_url": "https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/RegExp/hasIndices",
+            "support": {
+              "chrome": {
+                "version_added": "91"
+              },
+              "chrome_android": {
+                "version_added": "91"
```
Comment (ddbeck) [inline javascript/builtins/RegExp.json:437]:
```suggestion
                "version_added": "90"
```

**Review by ddbeck — CHANGES_REQUESTED** [review]
I was wondering how this was going and looked at the bug to see if it might've slipped. Instead, I think it's shipping quite soon. See inline comments for details. Thank you!

File: javascript/builtins/RegExp.json (hunk 1 above)
Comment (SebastianZ) [reply javascript/builtins/RegExp.json]: Done.

File: javascript/builtins/RegExp.json (hunk 2 above)
Comment (SebastianZ) [reply javascript/builtins/RegExp.json]: Done.

File: javascript/builtins/RegExp.json (hunk 3 above)
Comment (SebastianZ) [reply javascript/builtins/RegExp.json]: Done.

File: javascript/builtins/RegExp.json (hunk 4 above)
Comment (SebastianZ) [reply javascript/builtins/RegExp.json]: Done.

**Comment by SebastianZ** [comment]
Thank you, @ddbeck, for checking the compatibility and providing hints! I've fixed the information now.

File: javascript/builtins/RegExp.json [hunk 5]
```diff
@@ -426,6 +426,57 @@
...
+              },
+              "samsunginternet_android": {
+                "version_added": false
+              },
+              "webview_android": {
+                "version_added": "90"
+              }
+            },
+            "status": {
+              "experimental": true,
```
Comment (ddbeck) [inline javascript/builtins/RegExp.json:474]: Sorry, I ought to have caught this before. Generally, we set `experimental` to `false`, when there's two or more implementations.

```suggestion
              "experimental": false,
```

**Review by ddbeck — CHANGES_REQUESTED** [review]
Sorry, one last thing I didn't catch last time. Thank you again, @SebastianZ!

File: javascript/builtins/RegExp.json (hunk 5 above)
Comment (SebastianZ) [reply javascript/builtins/RegExp.json]: Thank you for the note! Changed it to `false`.

**Review by ddbeck — APPROVED** [review]
Looks perfect. Thank you! 🎉

**Pull request merged**